module/zfs/zvol.c

```diff
@@ -1217,8 +1217,6 @@ zvol_remove_minors_impl(const char *name)
 			    (task_func_t *)ops->zv_free, zv, TQ_SLEEP);
 			if (t == TASKQID_INVALID)
 				list_insert_head(&free_list, zv);
-			else
-				tid = t;
```

Comment:
According to the commit history, 899662e344b150037180a6f61acd186fec773c5f, the reason for adding this task queue dispatch was because `blk_cleanup_queue()` and `put_disk()` could take over 10ms.  Could you add a comment to `zvol_free()` in `module/os/linux/zfs/zvol_os.c` either above the whole function or above the `del_gendisk()` block to make it clear this hunk can be slow.

Reply:
> the reason for adding this task queue dispatch was because blk_cleanup_queue() and put_disk() could take over 10ms. 

Yea, and that's what I continue to see. In my case though, for example, I've see `blk_cleanup_queue` take over 256ms (in the 256ms to 512ms bucket of the histogram from BCC's `funclatency`).

E.g.
```
Function = blk_cleanup_queue
     msecs               : count     distribution
         0 -> 1          : 0        |                                        |
         2 -> 3          : 0        |                                        |
         4 -> 7          : 0        |                                        |
         8 -> 15         : 0        |                                        |
        16 -> 31         : 0        |                                        |
        32 -> 63         : 1        |*                                       |
        64 -> 127        : 7        |*********                               |
       128 -> 255        : 29       |****************************************|
       256 -> 511        : 14       |*******************                     |

Function = del_gendisk
     msecs               : count     distribution
         0 -> 1          : 0        |                                        |
         2 -> 3          : 0        |                                        |
         4 -> 7          : 0        |                                        |
         8 -> 15         : 1        |*                                       |
        16 -> 31         : 4        |******                                  |
        32 -> 63         : 22       |**************************************  |
        64 -> 127        : 23       |****************************************|
       128 -> 255        : 1        |*                                       |
```

I'll add a comment.

Reply:
I've added a small comment; please let me know if you think it's sufficient.

Reply:
That works for me.  Thanks.